### PR TITLE
sw-emulator-app: Lock mailbox using external interface.

### LINKED ISSUE #264: Bug : Emulator app can't load firmware image
After alignment between mailbox emulation behavior and RTL, emulator app can no longer load an image.

The specific change was to prevent the receiver from releasing the  mailbox lock.

### PR DESCRIPTION
**Pull request by korran**
Fixes #264.

### CODE REVIEW

File: sw-emulator/app/src/main.rs [hunk 1]
```diff
@@ -397,6 +395,8 @@ fn upload_fw_to_mailbox(mailbox: &mut MailboxInternal, firmware_buffer: Rc<Vec<u
     let mut soc = mailbox.as_external();
     // Write the cmd to mailbox.
 
+    assert_eq!(soc.read(RvSize::Word, OFFSET_LOCK).unwrap(), 0);
```
Comment (rusty1968) [inline sw-emulator/app/src/main.rs:398]: Is this for debug ?

Comment (rusty1968) [reply sw-emulator/app/src/main.rs:398]: No need to fix the firmware read complete in mailbox.rs?

**Review by rusty1968 — APPROVED** [review]

File: sw-emulator/app/src/main.rs (hunk 1 above)
Comment (rusty1968) [reply sw-emulator/app/src/main.rs:398]: Can we add Vishal's "make run" to the CI? We are not covering emulator app and we risk breaking it again.

**Review by rusty1968 — DISMISSED** [review]

File: sw-emulator/app/src/main.rs (hunk 1 above)
Comment (korran) [reply sw-emulator/app/src/main.rs:397]: Done

**Review by rusty1968 — APPROVED** [review]

**Pull request merged**